src/lib.rs

```diff
@@ -90,6 +89,8 @@
     while_true,
 )]
 
+#![cfg_attr(not(feature = "nightly"), deny(drop_with_repr_extern))]
```

Comment:
I didn't see this until I'd already written a different patch. This is smarter than what I did. I don't think we need to do this though, because in 6 weeks we'd have to remove it. Thanks though!